src/meta-srv/src/election/postgres.rs

```diff
@@ -35,50 +35,116 @@ use crate::error::{
 };
 use crate::metasrv::{ElectionRef, LeaderValue, MetasrvNodeInfo};
 
-// TODO(CookiePie): The lock id should be configurable.
-const CAMPAIGN: &str = "SELECT pg_try_advisory_lock({})";
-const STEP_DOWN: &str = "SELECT pg_advisory_unlock({})";
-// Currently the session timeout is longer than the leader lease time, so the leader lease may expire while the session is still alive.
-// Either the leader reconnects and step down or the session expires and the lock is released.
-const SET_IDLE_SESSION_TIMEOUT: &str = "SET idle_session_timeout = '10s';";
-
 // Separator between value and expire time.
 const LEASE_SEP: &str = r#"||__metadata_lease_sep||"#;
 
-// SQL to put a value with expire time. Parameters: key, value, LEASE_SEP, expire_time
-const PUT_IF_NOT_EXISTS_WITH_EXPIRE_TIME: &str = r#"
-WITH prev AS (
-    SELECT k, v FROM greptime_metakv WHERE k = $1
-), insert AS (
-    INSERT INTO greptime_metakv
-    VALUES($1, convert_to($2 || $3 || TO_CHAR(CURRENT_TIMESTAMP + INTERVAL '1 second' * $4, 'YYYY-MM-DD HH24:MI:SS.MS'), 'UTF8'))
-    ON CONFLICT (k) DO NOTHING
-)
+struct ElectionSqlFactory {
+    lock_id: u64,
+    table_name: String,
+}
 
-SELECT k, v FROM prev;
-"#;
+impl ElectionSqlFactory {
```

Comment:
How about modifying the factory to return a SqlTemplate? This way, we can avoid the calling `format!` with each invocation.